doc/events.rst

```diff
@@ -0,0 +1,98 @@
+Kansha events
+=============
+
+Definition
+----------
+
+Events are tuples in the form `(action, source, data)` where:
+
+* `action` is a string;
+* `source` is a list of business objects, each component relaying the event up must append itself to it; Is is a kind of path in reverse order;
+* `data` is data specific to the action.
+
+Events are returned as Nagare `answer` parameters by components.
+
+List of events
+--------------
+
+Actions end with verbs in base form most of the time. We don't mutate actions to conjugate them and mark they are done; that would not bring any improvement anyway.
+A business object who is concerned by an action knows what to do with it, othewise it just "bubbles" it up.
+
+The most significant actions may be bubbled after they are done too, to trigger some behavior in upper levels (e.g. history logging, indexing…). The action code is left untouched.
```

Comment:
"In general, the actions must always be bubbled to trigger some behavior in upper level (e.g. history logging, indexing…)"

~~The action code is left untouched~~